### PR TITLE
Add HasNoDuplicates, NthWeekOfMonth, HasNoDuplicates, IsMonotonicallyDecreasing, IsMonotonicallyIncreasing 

### LINKED ISSUE #2538: Add HasNoDuplicates, NthWeekOfMonth, HasNoDuplicates, IsMonotonicallyDecreasing, IsMonotonicallyIncreasing

### PR DESCRIPTION
**Pull request by gsheni**
- Fixes #2538
- Upgrade minimum pandas to 1.5.0 
- Removed the ``Correlation`` and ``AutoCorrelation`` primitive as these could lead to data leakage 
- Remove IntegerNullable support for ``Kurtosis`` primitive

### CODE REVIEW

**Comment by machineFL** [comment]
No linked issues found. Please add the corresponding issues in the pull request description. <br/> [Use GitHub automation to close the issue when a PR is merged](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

File: docs/source/release_notes.rst [hunk 1]
```diff
@@ -9,6 +9,8 @@ Future Release
         * Add ``MaxCount``, ``MedianCount``, ``MaxMinDelta``, ``NUniqueDays``, ``NMostCommonFrequency``,
             ``NUniqueDaysOfCalendarYear``, ``NUniqueDaysOfMonth``, ``NUniqueMonths``,
             ``NUniqueWeeks``, ``IsFirstWeekOfMonth`` (:pr:`2533`)
+        * Add ``HasNoDuplicates``, ``NthWeekOfMonth``, ``HasNoDuplicates``, ``IsMonotonicallyDecreasing``,
```
Comment (rwedge) [inline docs/source/release_notes.rst:12]: duplicate mention of `HasNoDuplicates`

File: featuretools/primitives/standard/aggregation/correlation.py
Comment (rwedge) [inline featuretools/primitives/standard/aggregation/correlation.py:1]: This deletion is not mentioned

File: featuretools/primitives/standard/aggregation/is_unique.py
Comment (rwedge) [inline featuretools/primitives/standard/aggregation/is_unique.py:1]: Not imported in module __init__ or mentioned in release notes

File: featuretools/tests/primitive_tests/aggregation_primitive_tests/test_agg_primitives.py [hunk 2]
```diff
@@ -12,6 +12,9 @@
     Correlation,
     DateFirstEvent,
     FirstLastTimeDelta,
+    HasNoDuplicates,
+    IsMonotonicallyDecreasing,
+    IsMonotonicallyIncreasing,
```
Comment (rwedge) [inline featuretools/tests/primitive_tests/aggregation_primitive_tests/test_agg_primitives.py:17]: should IsUnique be tested here as well?

**Review by dvreed77 — APPROVED** [review]
LGTM

**Pull request merged**